framework/wazuh/agent.py

```diff
@@ -556,14 +558,13 @@ def delete_groups(group_list=None):
                                       some_msg='Some groups were not deleted',
                                       none_msg='No group was deleted')
 
-    affected_agents = set()
     system_groups = get_groups()
     for group_id in group_list:
         try:
             # Check if group exists
             if group_id not in system_groups:
                 raise WazuhResourceNotFound(1710)
-            if group_id == 'default':
+            elif group_id == 'default':
                 raise WazuhError(1712)
             agent_list = list(map(operator.itemgetter('id'),
                                   WazuhDBQueryMultigroups(group_id=group_id, limit=None).run()['items']))
```

Comment:
This way is a little faster:
```suggestion
            agent_list = [agent['id'] for agent in WazuhDBQueryMultigroups(group_id=group_id, limit=None).run()['items']]
```